tests/contrib/pytest/test_pytest.py

```diff
@@ -394,3 +409,19 @@ def __repr__(self):
     decoded = json.loads(encoded)
     assert decoded["b"] == repr(dict_side_effect)
     assert decoded["c"] == repr(repr_side_effect)
+
+
+@pytest.mark.parametrize(
+    "repository_url,repository_name",
+    [
+        ("https://github.com/DataDog/dd-trace-py.git", "dd-trace-py"),
+        ("https://github.com/DataDog/dd-trace-py", "dd-trace-py"),
+        ("git@github.com:DataDog/dd-trace-py.git", "dd-trace-py"),
+        ("git@github.com:DataDog/dd-trace-py", "dd-trace-py"),
+        ("dd-trace-py", "dd-trace-py"),
+        ("git@hostname.com:org/repo-name.git", "repo-name"),
+        ("git@hostname.com:org/repo-name", "repo-name"),
```

Comment:
I left the comment on the commit not the PR 🤦🏻 

we should add an example with `ssh://` protocol, e.g. `ssh://git@hostname.com:org/repo-name`

Reply:
does it make sense to try other schemes like `git+ssh`?

Reply:
yes it does!

Reply:
@P403n1x87 do you have any examples of urls following the `git+ssh` protocol?

Reply:
@Yun-Kim see e.g. https://stackoverflow.com/questions/16584552/how-to-state-in-requirements-txt-a-direct-github-source